File: Java/MagicSquare.java
import java.util.*;
class  MagicSquare {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		System.out.println("Welcome to Magic Square !");
		System.out.println("Enter size. (size is odd number only.)");
		int n = scan.nextInt();
		int[][] m = new int[n+1][n+1];
		int r = 1;
		int c = n / 2 + 1;
		m[r][c] = 1;
		int n2 = n * n ;
		for (int k = 2; k <= n2; k++) {
			if (r == 1 && c == n) {
				r = 2;
			} else if (r ==1) {
				r = n;
				c++;
			} else if (c == n) {
				r--;
				c = 1;
			} else if (m[r-1][c+1] > 0) {
				r++;
			} else {
				r--;
				c++;
			}
			m[r][c] = k;
		}
		for (int i = 1; i <= n; i++) {
			for (int j = 1; j <= n; j++) {
				System.out.print(m[i][j] + "  ");
			}
			System.out.println(" ");
		}
		System.out.println("========== Program Terminated ==========");
	}
}

File: C/MAGIC_SQUARE.c
#include<stdio.h>
int main()
{
int n;

printf("enter the value of n to calculate the n*n magic square:");
scanf("%d",&n);
int a[n][n];
int i,j;
for(i=0;i<n;i++)
for(j=0;j<n;j++)
a[i][j]=0;

i=0;
j=n/2;
int k=n*n;
int p=1;
while(p<=k)
{
if(i<0&&j==n)
{
i+=2;
j--;
a[i][j]=p;
p++;
i--;j++;
}

else if(i<0&&j<n)
{
i=i+n;
a[i][j]=p;
p++;
i--;
j++;
}

else if(j==n&&i>=0)
{
j=0;
a[i][j]=p;
p++;
i--;
j++;
}

else if(a[i][j]!=0)
{
i+=2;
j--;
a[i][j]=p;
p++;
i--;j++;
}

else
{
a[i][j]=p;
p++;
i--;
j++;
}

}

for(i=0;i<n;i++)
{
for(j=0;j<n;j++)
{printf("%d",a[i][j]);
}
printf("\n");

}

}
